Fixed behavior for prompts having a when-function

koppor opened the pull request:
Follow-up to https://github.com/artefact-group/yeoman-option-or-prompt/pull/7:

Simplified solution to fix #6.

- Closes https://github.com/artefact-group/yeoman-option-or-prompt/pull/7
- Fixes #6

nur-psoehnlein committed:
Fixed behavior for prompts having a when-function.

Fixes artefact-group/yeoman-option-or-prompt#6

koppor committed:
Merge remote-tracking branch 'philippsoehnlein/master' into fix-when

# Conflicts:
#	index.js
#	package.json

koppor committed:
Streamline code

funwhilelost reviewed (approved):
I like it.  Let me see if I still have access to the npm credentials.

koppor commented:
Give me time for an update. I was too fast summiting the change 😅

koppor committed:
Fix return value (and indent)

koppor committed:
Fix method usage in README.md

koppor commented:
I included the fixes. Note that the method `return`s the properties (also before the patch). Thus, I updated the signature in the README.md. In my code, I did not need `bind(this)`, thus, I removed it in README.md, too.

koppor commented:
Need to go to draft, because "normal" prompting stopped working.

koppor committed:
Fix prompting

koppor commented:
Should be fixed now :)

koppor commented:
Stopped working while refactoring the surrounding code. Investigating.

koppor committed:
Fix choices creation

koppor commented:
It's live in my project - and works

- Patch: https://github.com/latextemplates/generator-latex-template/blob/main/patches/yeoman-option-or-prompt%2B2.0.1.patch
- Automatically patched via [patch-package](https://www.npmjs.com/package/patch-package)

koppor commented:
Note, I am still with yeoman-generator v5, because I didn't have time to investigate the consequences of their change to TypeScript.

funwhilelost commented:
@koppor Thanks for your work on this.  I haven't been able to get access back to this repo so I guess it's abandoned.
